src/wp-includes/class-wp-theme.php

```diff
@@ -2048,6 +2048,10 @@ public static function network_disable_theme( $stylesheets ) {
 		$allowed_themes = get_site_option( 'allowedthemes' );
 		foreach ( $stylesheets as $stylesheet ) {
 			if ( isset( $allowed_themes[ $stylesheet ] ) ) {
+				$theme = wp_get_theme( $stylesheet );
```

Comment:
Should we also cache the theme patterns on multisite when `network_enable_theme()` is called?

Reply:
Separate question related to this new change: Why does the pattern cache need to be cleared when disabling the theme network-wide? Disabling isn't the same as deactivating. Will this actually deactivate the theme even for sites that may have it as the active theme?

I'm not sure this is needed given the invalidation already happens on theme deactivation.

Reply:
@spacedmonkey I have tested and verified this: Network disabling a theme doesn't impact theme activation at all. If a site has that theme activated while it is network disabled, the site will still have that theme active. So I think clearing the cache here is irrelevant.